src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,720 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
-                getTurnDirection(intersection[obvious_turn_index].angle);
+            intersection[*obvious].instruction.type = TurnType::NewName;
+            intersection[*obvious].instruction.direction_modifier =
+                getTurnDirection(intersection[*obvious].angle);
         }
         else
         {
-            intersection[obvious_turn_index].instruction.type = TurnType::Suppressed;
+            intersection[*obvious].instruction.type = TurnType::Suppressed;
         }
     }
 
+    if (sliproad_found)
+    {
+        geojson_points.Write(std::vector<NodeID>{b, c, d});
+        geojson_lines.Write(std::vector<NodeID>{b, d});
+    }
+
     return intersection;
 }
 
+// Implementation details
+
+// Skips over `tl` traffic light and returns
+//
+//  a ... tl ... b .. c
+//               .
+//               .
+//               d
+//
+//  ^ at
+//     ^ road
+boost::optional<SliproadHandler::IntersectionAndNode>
+SliproadHandler::getNextIntersection(const NodeID at, const ConnectedRoad &road) const
+{
+    auto intersection = intersection_generator(at, road.eid);
+    auto in_edge = road.eid;
+    auto intersection_node = node_based_graph.GetTarget(in_edge);
+
+    // To prevent ending up in an endless loop, we remember all visited nodes. This is
+    // necessary, since merging of roads can actually create enterable loops of degree two
+    std::unordered_set<NodeID> visited_nodes;
+
+    auto node = at;
+    while (intersection.size() == 2 && visited_nodes.count(node) == 0)
+    {
+        visited_nodes.insert(node);
+        node = node_based_graph.GetTarget(in_edge);
+
+        // We ended up in a loop without exit
+        if (node == at)
+        {
+            return boost::none;
+        }
+
+        in_edge = intersection[1].eid;
+        intersection = intersection_generator(node, in_edge);
+        intersection_node = node_based_graph.GetTarget(in_edge);
+    }
+
+    if (intersection.size() <= 2)
+    {
+        return boost::none;
+    }
+
+    return boost::make_optional(IntersectionAndNode{intersection, intersection_node});
+}
+
+boost::optional<std::size_t> SliproadHandler::getObviousIndexWithSliproads(
+    const EdgeID from, const Intersection &intersection, const NodeID at) const
+{
+    BOOST_ASSERT(from != SPECIAL_EDGEID);
+    BOOST_ASSERT(at != SPECIAL_NODEID);
+
+    // If a turn is obvious without taking Sliproads into account use this
+    const auto index = findObviousTurn(from, intersection);
+
+    if (index != 0)
+        return boost::make_optional(index);
+
+    // Otherwise check if the road is forking into two and one of them is a Sliproad;
+    // then the non-Sliproad is the obvious one.
+    if (intersection.size() != 3)
+        return boost::none;
+
+    const auto forking = intersection[1].instruction.type == TurnType::Fork &&
+                         intersection[2].instruction.type == TurnType::Fork;
+
+    if (!forking)
```

Comment:
am I reading this correctly? You rely on forks to have a obvious index? That sounds strange to me.